### PR TITLE
PWGHF: adapt toXiPi tasks for ambig tracks and PV refit, use TrackParCov and minor changes

### PR DESCRIPTION
**Pull request by ZFederica**
- implement changes for ambiguous tracks
- implement changes for PV refit
- switch to TrackParCov to create tracks
- add dcaZ info
- fix event counter

### CODE REVIEW

File: PWGHF/TableProducer/candidateCreatorToXiPi.cxx [hunk 1]
```diff
@@ -77,12 +80,18 @@ struct HfCandidateCreatorToXiPi {
 
   // filter to use only HF selected collisions
   Filter filterSelectCollisions = (aod::hf_sel_collision::whyRejectColl == 0);
+  Filter filterSelectTrackIds = (aod::hf_sel_track::isSelProng >= 4);
 
   using SelectedCollisions = soa::Filtered<soa::Join<aod::Collisions, aod::HfSelCollision>>;
-  using MyTracks = soa::Join<aod::BigTracks, aod::TracksDCA, aod::TrackSelection>;
+  using MyTracks = soa::Join<aod::BigTracks, aod::TracksDCA, aod::HfPvRefitTrack>;
+  using FilteredHfTrackAssocSel = soa::Filtered<soa::Join<aod::TrackAssoc, aod::HfSelTrack>>; // HfTrackAssoc
```
Comment (vkucera) [inline PWGHF/TableProducer/candidateCreatorToXiPi.cxx:87]: Move `using` before filters.

File: PWGHF/TableProducer/candidateCreatorToXiPi.cxx [hunk 2]
```diff
@@ -94,251 +103,309 @@ struct HfCandidateCreatorToXiPi {
...
+          // check not to take the same particle twice in the decay chain
+          if (trackPion.globalIndex() == trackXiDauCharged.globalIndex() || trackPion.globalIndex() == trackV0Dau0.globalIndex() || trackPion.globalIndex() == trackV0Dau1.globalIndex() || trackPion.globalIndex() == casc.globalIndex()) {
+            continue;
+          }
 
-        // check not to take the same particle twice in the decay chain
-        if (trackPion.globalIndex() == trackXiDauCharged.globalIndex() || trackPion.globalIndex() == trackV0Dau0.globalIndex() || trackPion.globalIndex() == trackV0Dau1.globalIndex() || trackPion.globalIndex() == casc.globalIndex()) {
-          continue;
-        }
+          // pseudirapidity
```
Comment (vkucera) [inline PWGHF/TableProducer/candidateCreatorToXiPi.cxx:253]:
```suggestion
          // pseudorapidity
```

**Review by vkucera — CHANGES_REQUESTED** [review]

File: PWGHF/TableProducer/candidateCreatorToXiPi.cxx (hunk 1 above)
Comment (ZFederica) [reply PWGHF/TableProducer/candidateCreatorToXiPi.cxx:87]: Ok

File: PWGHF/DataModel/CandidateReconstructionTables.h [hunk 3]
```diff
@@ -1478,10 +1481,12 @@ DECLARE_SOA_COLUMN(EtaPiFromOme, etaPiFromOme, double);
 DECLARE_SOA_COLUMN(EtaOmegac, etaOmegac, double);
 DECLARE_SOA_COLUMN(EtaCascade, etaCascade, double);
 DECLARE_SOA_COLUMN(EtaV0, etaV0, double);
-DECLARE_SOA_COLUMN(DcaXYToPVPrimaryPi, dcaXYToPVPrimaryPi, double);
 DECLARE_SOA_COLUMN(DcaXYToPVV0Dau0, dcaXYToPVV0dau0, double);
 DECLARE_SOA_COLUMN(DcaXYToPVV0Dau1, dcaXYToPVV0dau1, double);
 DECLARE_SOA_COLUMN(DcaXYToPVCascDau, dcaXYToPVCascdau, double);
+DECLARE_SOA_COLUMN(DcaZToPVV0Dau0, dcaZToPVV0dau0, double);
+DECLARE_SOA_COLUMN(DcaZToPVV0Dau1, dcaZToPVV0dau1, double);
+DECLARE_SOA_COLUMN(DcaZToPVCascDau, dcaZToPVCascdau, double);
```
[the remark this answers is not in the document]
Comment (ZFederica) [reply PWGHF/DataModel/CandidateReconstructionTables.h]: Ok. Float is enough, changed.

[the remark this answers is not in the document]
Comment (vkucera) [reply PWGHF/DataModel/CandidateReconstructionTables.h]: OK, thanks. What about the other columns?

File: PWGHF/TableProducer/candidateCreatorToXiPi.cxx (hunk 1 above)
Comment (vkucera) [reply PWGHF/TableProducer/candidateCreatorToXiPi.cxx]: Maybe there was a misunderstanding. All the `using` declarations should be before all the filters. (See the HF guidelines.)

Comment (ZFederica) [reply PWGHF/TableProducer/candidateCreatorToXiPi.cxx]: Even if a filter is then to be used inside the using?

File: PWGHF/DataModel/CandidateReconstructionTables.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (ZFederica) [reply PWGHF/DataModel/CandidateReconstructionTables.h]: The other are necessarily double

File: PWGHF/TableProducer/candidateCreatorToXiPi.cxx (hunk 1 above)
Comment (vkucera) [reply PWGHF/TableProducer/candidateCreatorToXiPi.cxx]: That cannot happen. Filters are not used anywhere in the code once they are declared.

Comment (ZFederica) [reply PWGHF/TableProducer/candidateCreatorToXiPi.cxx]: Ok

**Comment by ZFederica** [comment]
@vkucera title changed

**Review by vkucera — CHANGES_REQUESTED** [review]
Some leftovers

**Review by vkucera — APPROVED** [review]

**Pull request merged**